AberrantSMPP/Packet/Pdu.cs

```diff
@@ -163,7 +165,7 @@ protected Pdu(byte[] incomingBytes)
 			_CommandID = DecodeCommandId(_PacketBytes);
 			_CommandStatus = (CommandStatus)UnsignedNumConverter.SwapByteOrdering(BitConverter.ToUInt32(_PacketBytes, 8));
 			_SequenceNumber = UnsignedNumConverter.SwapByteOrdering(BitConverter.ToUInt32(_PacketBytes, 12));
-			_PacketBytes = TrimResponsePdu(_PacketBytes);
+			_PacketBytes = TrimResponsePdu(new Queue<byte>(_PacketBytes));
```

Comment:
avoid this, by simply passing _PacketBytes to TrimResponsePdu. And also let's add a Guard against _PacketBytes.Length != _CommandLength..

Reply:
I deleted the call to TrimResponsePdu call in the PDU ctor, in this way we use _PacketBytes (the bytes that we receive) directly, i added also an if with the same validation instead of a Guard.